tests/OSModulesTest.php

```diff
@@ -101,6 +101,7 @@ public function testOS($os, $variant, $modules): void
 
         try {
             Debug::set(false); // avoid all undefined index errors in the legacy code
+            $modules = array_fill_keys(array_intersect(array_keys($this->discoveryModules), array_keys($modules)), true);
```

Comment:
Well @laf, we still don't get why this exact line here is necessary.
But you can review the code (in Cisco.php and cisco-entity-sensor.inc.php), which are the useful parts of this PR :)

Reply:
@PipoCanaja The code looks ok to me. Other than the test data I don't have access to any devices this impacts so I can't check any further.

I'd be happy to merge as is and thanks for all the effort you've put into it.

@librenms/core anyone else?

Reply:
I'll try to do something about that next WE. When do you plan next release ? I would have liked to make it but it will be a little short I suppose.

Reply:
What about just linking to this PR as a comment against that line in the code @PipoCanaja for future reference.